include/aspect/compat.h

```diff
@@ -28,6 +28,353 @@
+  class InterpolatedUniformGridData : public dealii::Function<dim>
+  {
+    public:
+      /**
+       * Constructor
+       * @param interval_endpoints The left and right end points of the
+       * (uniformly subdivided) intervals in each of the coordinate directions.
+       * @param n_subintervals The number of subintervals in each coordinate
+       * direction. A value of one for a coordinate means that the interval is
+       * considered as one subinterval consisting of the entire range. A value
+       * of two means that there are two subintervals each with one half of the
+       * range, etc.
+       * @param data_values A dim-dimensional table of data at each of the mesh
+       * points defined by the coordinate arrays above. Note that the Table
+       * class has a number of conversion constructors that allow converting
+       * other data types into a table where you specify this argument.
+       */
+      InterpolatedUniformGridData(
+        const std::array<std::pair<double, double>, dim> &interval_endpoints,
+        const std::array<unsigned int, dim>              &n_subintervals,
+        const Table<dim, double>                         &data_values);
+
+      /**
+       * Compute the value of the function set by bilinear interpolation of the
+       * given data set.
+       *
+       * @param p The point at which the function is to be evaluated.
+       * @param component The vector component. Since this function is scalar,
+       * only zero is a valid argument here.
+       * @return The interpolated value at this point. If the point lies outside
+       * the set of coordinates, the function is extended by a constant.
+       */
+      virtual
+      Tensor<1, dim>
+      gradient(const Point<dim> &p,
+               const unsigned int component = 0) const override;
+
+      /**
+       * Compute the value of the function set by bilinear interpolation of the
+       * given data set.
+       *
+       * @param p The point at which the function is to be evaluated.
+       * @param component The vector component. Since this function is scalar,
+       * only zero is a valid argument here.
+       * @return The interpolated value at this point. If the point lies outside
+       * the set of coordinates, the function is extended by a constant.
+       */
+      virtual
+      double
+      value(const Point<dim> &p,
+            const unsigned int component = 0) const override;
+
+
+    private:
+      /**
+       * The set of interval endpoints in each of the coordinate directions.
+       */
+      const std::array<std::pair<double, double>, dim> interval_endpoints;
+
+      /**
+       * The number of subintervals in each of the coordinate directions.
+       */
+      const std::array<unsigned int, dim> n_subintervals;
+
+      /**
+       * The data that is to be interpolated.
+       */
+      const Table<dim, double> data_values;
+
+  };
+}
+
+namespace aspect
+{
+  DEAL_II_DISABLE_EXTRA_DIAGNOSTICS
```

Comment:
what warnings are you suppressing?

Reply:
this is to suppress "functions unused" warnings. We don't know exactly why they appear - the functions are definitely used.

Reply:
can you try marking your functions ``inline`` instead?